### PR TITLE
fix: add ingress annotation to each session so iframes work

### LINKED ISSUE #793: Add nginx ingress annotation through a patch
Since we embed the sessions into a renkulab pages, it would make sense to add the following ingress annotation to the ingress objects which are created as part of the JupyterServer sessions: 
```
    nginx.ingress.kubernetes.io/configuration-snippet: |
        more_set_headers "Content-Security-Policy: frame-ancestors 'self' {{ .Values.ui.baseUrl }}"; 
```
Although we have already addressed this issue at the level of the session base images, adding it here would be more generic and independent of what's running inside the actual session.

### PR DESCRIPTION
**Pull request by olevski**
Fixes #793 

/deploy

### CODE REVIEW

File: renku_notebooks/api/classes/server.py [hunk 1]
```diff
@@ -727,6 +727,23 @@ def _get_session_manifest(self):
                     ],
                 }
             )
+        patches.append(
+            {
+                "type": "application/json-patch+json",
+                "patch": [
+                    {
+                        "op": "add",
+                        "path": "/ingress/metadata/annotations/-",
```
Comment (ableuler) [inline renku_notebooks/api/classes/server.py:736]: Isn't that the syntax for adding to an array? The annotations are a map iirc.

Comment (olevski) [reply renku_notebooks/api/classes/server.py:736]: Ah yeah that is true. My bad.

File: renku_notebooks/api/classes/server.py [hunk 2]
```diff
@@ -727,6 +727,25 @@ def _get_session_manifest(self):
                     ],
                 }
             )
+        patches.append(
+            {
+                "type": "application/json-patch+json",
+                "patch": [
+                    {
+                        "op": "add",
+                        # "~1" == "/" for rfc6902 json patches
```
Comment (ableuler) [inline renku_notebooks/api/classes/server.py:736]: ouff, good find!

**Review by ableuler — APPROVED** [review]

**Comment by olevski** [comment]
p.s. The acceptance tests have been failing on launching anonymous sessions. But I tested this PRs deployment and I can launch anonymous sessions without a problem.

**Review by ableuler — APPROVED** [review]
LGTM 👍

**Pull request merged**